data_scrapers/sonoma_county.py

```diff
@@ -0,0 +1,216 @@
+            raise FutureWarning('The transmission type {0} was not found in transmission_type_conversion'.format(type))
+        type = transmission_type_conversion[type]
+        transmissions[type] = int(number)
+    return transmissions
+
+def transform_tests(tests_tag: element.Tag) -> Dict[str, int]:
+    tests = {}
+    rows = get_rows(tests_tag)
+    for row in rows:
+        result, number, _pct = get_cells(row)
+        lower_res = result.lower()
+        tests[lower_res] = int(number.replace(',', ''))
+    return tests;
+
+def generic_transform(tag: element.Tag) -> Dict[str, int]:
+    """
+    Transform function for tables which don't require any special processing.
+    Takes in a BeautifulSoup tag for a table and returns a dictionary
+    in which the keys are strings and the values integers
+    """
+    categories = {}
+    rows = get_rows(tag)
+    for row in rows:
+        cat, cases, _pct = get_cells(row)
+        categories[cat] = int(cases)
+    return categories
+
+def get_unknown_race(race_eth_tag: element.Tag) -> int:
+    """
+    Gets the notes under the 'Cases by race and ethnicity' table to find the
+    number of cases where the person's race is unknown
+    """
+    parent = race_eth_tag.parent
+    note = parent.find('p').text
+    matches = re.search('(\d+) \(\d{1,3}%\) missing race/ethnicity', note)
+    if not matches:
+        raise FutureWarning('The format of the note with unknown race data has changed')
+    return(int(matches.groups()[0]))
+
+def transform_race_eth(race_eth_tag: element.Tag) -> Dict[str, int]:
+    """
+    Takes in the BeautifulSoup tag for the cases by race/ethnicity table and
+    transforms it into an object of form:
+    'race_eth': {'Asian': -1, 'Latinx_or_Hispanic': -1, 'Other': -1, 'White':-1, 'Unknown': -1}
+    NB: These are the only races reported seperatley by Sonoma county at this time
+    """
+    race_cases = {}
+    race_transform = {'Asian/Pacific Islander, non-Hispanic': 'Asian', 'Hispanic/Latino': 'Latinx_or_Hispanic', 'Other*, non-Hispanic': 'Other', 'White, non-Hispanic': 'White'}
+    rows = get_rows(race_eth_tag)
+    for row in rows:
+        group_name, cases, _pct = get_cells(row)
+        if group_name not in race_transform:
+            raise FutureWarning('The racial group {0} is new in the data -- please adjust the scraper accordingly')
+        internal_name = race_transform[group_name]
+        race_cases[internal_name] = int(cases)
+    race_cases['Unknown'] = get_unknown_race(race_eth_tag)
+    return race_cases
+
+def transform_total_hospitalizations(hospital_tag: element.Tag) -> Dict[str, int]:
+    """
+    Takes in a BeautifulSoup tag of the cases by hospitalization table and
+    returns a dictionary with the numbers of hospitalized and non-hospitalized
+    cases
+    """
+    hospitalizations = {}
+    rows = get_rows(hospital_tag)
+    for row in rows:
+        hospitalized, number, _pct = get_cells(row)
+        if hospitalized == 'Yes':
```

Comment:
Maybe worth a case-insensitive comparison, just in case this changes in the future.

Reply:
I thought we would use the [chhs](https://data.chhs.ca.gov/dataset/california-covid-19-hospital-data-and-case-statistics/resource/6cd8d424-dfaa-4bdd-9410-a3d656e1176e?inner_span=True) as a common source for all hospitalization records. It looks like this is where most counties are getting their information, and we can be sure to have some sort of standard across counties by going straight to the chhs API. This referenced in issue #29. This is why the data_model doesn't currently include a hospitalization sub-table, I thought we could add it separately from the individual county scrapers. I decided to exclude hospitalization data available through SF county, to avoid confusion when we do get to #29.